ros/src/waypoint_updater/waypoint_updater.py

```diff
@@ -94,12 +96,15 @@ def find_next_waypoint(self):
             ego_pose = self.ego.pose
             n_waypoints = len(self.waypoints)
             # Find the closest waypoint
-            for i in range(n_waypoints):
-                wp_pos = self.waypoints[i].pose.pose.position
+            for i in range(self.next_idx, self.next_idx + n_waypoints):
+                idx = (i + n_waypoints) % n_waypoints
```

Comment:
Hmm I don't quite understand this: `idx = (i + n_waypoints) % n_waypoints`, why is it needed?

Reply:
This way we will avoid negative indices, `i` starts as `-1`.

Reply:
Fair enough! :)